### PR TITLE
User proxy and CLI user argument

### PR DESCRIPTION
**Pull request by robinschmid**
adds to PR #1861 I think it is easier to review together

Adds a -user argument which wants a user file as input.

also adds a -login argument that forces a user login + download of the user file.

### CODE REVIEW

File: mzmine-community/src/main/java/io/github/mzmine/main/MZmineCore.java [hunk 1]
```diff
@@ -188,35 +189,59 @@ public static void main(final String[] args) {
...
-                  """, ButtonType.YES, ButtonType.NO);
-              if (btn == ButtonType.YES) {
-                ConfigService.getConfiguration().getPreferences().showSetupDialog(true, "Proxy");
-              }
-            }
-          });
+        if (mzEvent instanceof ProxyChangedEvent pevent) {
+          ConfigService.getPreferences().setProxy(pevent.proxy());
         }
+//        if (mzEvent instanceof AuthServerNotReachedEvent) {
```
Comment (SteffenHeu) [inline mzmine-community/src/main/java/io/github/mzmine/main/MZmineCore.java:203]: remove

File: mzmine-community/src/main/java/io/github/mzmine/main/MZmineCore.java [hunk 2]
```diff
@@ -188,35 +189,59 @@ public static void main(final String[] args) {
           if (DesktopService.isGUI()) {
             getDesktop().addTab(UsersTab.showTab());
           } else {
-            getDesktop().displayMessage("Requires user login. Open mzmine and login to a user");
+            try {
+              new UsersController().onLoginOrRegister(LoginOptions.CONSOLE);
```
Comment (SteffenHeu) [inline mzmine-community/src/main/java/io/github/mzmine/main/MZmineCore.java:193]: should this be handled by the current user service?

File: mzmine-community/src/main/java/io/github/mzmine/main/MZmineCore.java [hunk 3]
```diff
@@ -228,6 +253,17 @@ public static void main(final String[] args) {
       // apply memory management option
       keepInMemory.enforceToMemoryMapping();
 
+      // login user if cli option
+      if (argsParser.isCliLogin()) {
+        try {
+          logger.info("CLI user login");
+          new UsersController().loginOrRegisterConsoleBlocking();
```
Comment (SteffenHeu) [inline mzmine-community/src/main/java/io/github/mzmine/main/MZmineCore.java:260]: see above

**Review by SteffenHeu — COMMENTED** [review]
looks good, some slight comments

File: mzmine-community/src/main/java/io/github/mzmine/main/MZmineCore.java (hunk 2 above)
Comment (robinschmid) [reply mzmine-community/src/main/java/io/github/mzmine/main/MZmineCore.java:193]: Yeah I think its the best way to do things its a single point of entry and the controller streamlines the whole user things

Comment (robinschmid) [reply mzmine-community/src/main/java/io/github/mzmine/main/MZmineCore.java:193]: At least feels good to use

**Pull request merged**